src/molecule/config.py

```diff
@@ -240,6 +258,20 @@ def molecule_directory(self) -> str:
         """
         return molecule_directory(self.project_directory)
 
+    @cached_property
+    def collection(self) -> CollectionData | None:
+        """Collection metadata sourced from galaxy.yml.
+
+        Returns:
+            A dictionary of information about the collection molecule is running inside, if any.
+        """
+        if not self.collection_directory:
+            return None
+
+        galaxy_file = self.collection_directory / "galaxy.yml"
+        galaxy_data = util.safe_load_file(galaxy_file)
```

Comment:
If we don't at least have name and namespace we should fail fast and let the user know the galaxy.yml file needs these values.  I think the intent of the user here would be to be collection aware rather than not